src/Illuminate/Queue/Worker.php

```diff
@@ -208,8 +212,14 @@ public function process($connection, Job $job, $maxTries = 0, $delay = 0)
 			return ['job' => $job, 'failed' => false];
 		}
 
-		catch (\Exception $e)
+		catch (Exception $e)
 		{
+			// If the job failed and hit the max, let's log it and exit the process early.
+			if ($maxTries > 0 && ($job->attempts() + 1) >= $maxTries)
```

Comment:
Looking at this check right here. Say my job is on attempt number 2 and max tries is 3. If my job throws an exception and you check attempts() + 1 it's going to log it as a failed job when really it only has 2 attempts?